# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsFrameLoader.cpp
# Commit: aea4783f8b48
# Full Hash: aea4783f8b4867167607f8ba00eadc4aaa45d61d
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2020-02-07 03:55:51
# Regressor Bug: 1582832
# File Overlap Count: 1
# Description:
#   Bug 1582832: Part 3 - Correctly handle windowless BCs created for printing. r=nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D61006
# ==============================================================================

diff -r cefac6461948 -r aea4783f8b48 dom/base/nsFrameLoader.cpp
--- a/dom/base/nsFrameLoader.cpp	Thu Feb 06 19:08:04 2020 +0000
+++ b/dom/base/nsFrameLoader.cpp	Thu Feb 06 19:08:06 2020 +0000
@@ -289,8 +289,19 @@
   // for the BrowsingContext, and cause no end of trouble.
   if (IsTopContent(parentContext, aOwner)) {
     // Create toplevel content without a parent & as Type::Content.
-    return BrowsingContext::CreateDetached(nullptr, aOpener, frameName,
-                                           BrowsingContext::Type::Content);
+    RefPtr<BrowsingContext> bc = BrowsingContext::CreateDetached(
+        nullptr, aOpener, frameName, BrowsingContext::Type::Content);
+
+    // If this is a mozbrowser frame, pretend it's windowless so that it gets
+    // ownership of its BrowsingContext even though it's a top-level content
+    // frame. This is horrible, but will fortunately go away soon.
+    if (nsCOMPtr<nsIMozBrowserFrame> mozbrowser =
+            aOwner->GetAsMozBrowserFrame()) {
+      if (mozbrowser->GetReallyIsBrowser()) {
+        bc->SetWindowless();
+      }
+    }
+    return bc.forget();
   }
 
   auto type = parentContext->IsContent() ? BrowsingContext::Type::Content